### DISCUSSION TITLE
How to get evp_md_ctx_st.algctx in openssl v3?

### DISCUSSION THREAD

**Discussion by TheBigFish** [openssl/openssl#24501] [Q&A] [answered]
Hi, 

The legacy version digest using `evp_md_ctx_st.md_data` to save the digest context，for example ,the `SM3_CTX` data。I can use the `EVP_MD_CTX_md_data` to get the internal context data。
When using provider，the context data move to `evp_md_ctx_st.algctx`，and there is no method to get the context。My question  is how to get the internal context data in provider mode?

thanks.

**Comment by t8m** [comment, marked as answer]
I assume you want to save the internal digest state to be able to restore it later. Unfortunately this is not implemented for providers.
There is an existing RFE: https://github.com/openssl/openssl/issues/14222
